File: Society.java
import java.util.ArrayList;

public class Society {
    private String name;
    private ArrayList<Member> members;

    public Society(String name) {
        this.name = name;
        this.members = new ArrayList<>();
    }

    public void addStandardMember(String name, String address) {
        members.add(new StandardMember(name, address));
    }

    public void addSeniorMember(String name, String address, int fee) {
        members.add(new SeniorMember(name, address, fee));
    }

    public int getFeeTotal() {
        int totalFees = 0;
        for (Member member : members) {
            totalFees += member.getFee();
        }
        return totalFees;
    }

    public void printAllMembers() {
        for (int i = 0; i < members.size(); i++) {
            Member mb = members.get(i);
            System.out.println("i = " + i + " Name = " + mb.getName() +
                    " Address = " + mb.getAddress() + " Fees = " + mb.getFee());
        }
    }
    public String getName(){
        return this.name;
    }
}

File: society.py
from standard_member import StandardMember
from senior_member import SeniorMember

class Society:
    def __init__(self, name):
        self.name = name
        self.members = []  # Python list

    def add_standard_member(self, name, address):
        self.members.append(StandardMember(name, address))

    def add_senior_member(self, name, address, fee):
        self.members.append(SeniorMember(name, address, fee))

    def get_fee_total(self):
        total_fees = 0
        for member in self.members:
            total_fees += member.get_fee()
        return total_fees

    def print_all_members(self):
        for i, member in enumerate(self.members):
            print(f"i = {i} {member}")
            #print(f"i = {i} Name = {member.get_name()} Address = {member.get_address()} Fees = {member.get_fee()}")

    def get_name(self):
        return self.name